components/cop_codegen/src/lib.rs

```diff
@@ -36,3 +43,19 @@ pub fn aggr_function_derive(input: TokenStream) -> TokenStream {
         Ok(r) => TokenStream::from(r.generate_tokens()),
     }
 }
+
+#[proc_macro_attribute]
+pub fn rpn_fn(attr: TokenStream, input: TokenStream) -> TokenStream {
+    let meta_parser = Punctuated::<Ident, Token![,]>::parse_terminated;
+    let meta = meta_parser.parse(attr).unwrap().into_iter().collect();
+    let item_fn = parse_macro_input!(input as ItemFn);
+
+    let input = RpnFnGenerator::new(meta, item_fn);
+    match input.map(RpnFnGenerator::generate) {
+        Ok(tokens) => TokenStream::from(tokens),
+        Err(e) => {
+            e.emit();
+            std::process::exit(1)
```

Comment:
Could you return an empty TokenStream or something like that here?

Reply:
emm...What's the benefit of returning an empty TokenStream?

Reply:
It allows the compiler collect as many errors as possible. See also https://github.com/rust-lang/rust/issues/54140#issuecomment-434969649